aten/src/THC/THCTensorMathPairwise.cu

```diff
@@ -232,22 +154,18 @@ struct TensorRemainderOp<double> {
 };
 
 template <>
-struct TensorRemainderOp<half> {
-  TensorRemainderOp(half v): fval(THC_half2float(v)) {}
+struct TensorRemainderOp<THHalf> {
+  TensorRemainderOp(THHalf v): val(v) {}
 
-  __device__ __forceinline__ void operator()(half* out, half* in) {
-    float fin = __half2float(*in);
-    float fout = fin - fval * floorf(fin / fval);
-    *out = __float2half(fout);
+  __device__ __forceinline__ void operator()(THHalf* out, THHalf* in) {
+    *out = *in - val * floorf(*in / val);
```

Comment:
Err, it might not matter, but you're not actually converting input to float anymore before dividing!